nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,278 @@
+        }
+      '';
+      description = ''
+        Draupnir settings (see [Draupnir's default configuration](https://github.com/the-draupnir-project/Draupnir/blob/main/config/default.yaml) for available settings).
+        These settings will override settings made by the module config.
+      '';
+      default = { };
+      type = lib.types.submodule {
+        freeformType = format.type;
+        options = {
+          #region Readonly settings - these settings are not configurable
+          dataPath = lib.mkOption {
+            type = lib.types.str;
+            default = "/var/lib/draupnir";
+            readOnly = true;
+            description = ''
+              The path where Draupnir stores its data.
+
+              ::: {.note}
+                If you want to customize where this data is stored, use a bind mount.
+              :::
+            '';
+          };
+
+          pantalaimon = lib.mkOption {
+            readOnly = true;
+            description = ''
+              `pantalaimon` settings (enables E2E Encryption support).
+              This property is read-only, please configure `services.draupnir.pantalaimon` instead!
+            '';
+            default = {};
+            type = lib.types.submodule {
+              freeformType = format.type;
+              options = {
+                use = lib.mkOption {
+                  type = lib.types.bool;
+                  default = cfg.pantalaimon.enable;
+                  defaultText = "cfg.pantalaimon.enable";
+                  readOnly = true;
+                  description = ''
+                    Whether to use `pantalaimon` for E2E encryption. Enabled if `services.draupnir.pantalaimon.enable` is `true`.
+                  '';
+                };
+                username = lib.mkOption {
+                  type = lib.types.str;
+                  default = cfg.pantalaimon.username;
+                  defaultText = "cfg.pantalaimon.username";
+                  readOnly = true;
+                  description = ''
+                    Account name on the Matrix homeserver. Configured in `services.draupnir.pantalaimon.username`.
+                  '';
+                };
+              };
+            };
+          };
+          #endregion
+
+          #region Base settings
+          homeserverUrl = lib.mkOption {
+            type = lib.types.str;
+            defaultText = ''
+              URI to the pantalaimon instance, if configured. Else unset.
+            '';
+            description = ''
+              Base URL of the Matrix homeserver, that provides the Client-Server API.
+              If `services.draupnir.pantalaimon.enable` is `true`, this option will become read only. Configure `services.draupnir.pantalaimon.options.homeserver` instead in that case.
+              The listen address of `pantalaimon` will then become the `homeserverUrl` of `draupnir`.
+            '';
+          } // lib.optionalAttrs cfg.pantalaimon.enable {
```

Comment:
This is incorrect and is causing your ci failures. You want to change it to directly set it (probably *without* mkDefault, since the docs here suggest it shouldn't be set by the user in such a case) in config =, instead of putting it in here.

Should be something like: 
`homeserverUrl = lib.mkIf cfg.pantalaimon.enable (the default here)`